dart/test/http_client/sentry_http_client_test.dart

```diff
@@ -0,0 +1,236 @@
+          .captured
+          .single as Breadcrumb;
+
+      expect(breadcrumb.type, 'https');
+      expect(breadcrumb.data, <String, dynamic>{
+        'url': 'https://example.com',
+        'method': 'POST',
+        'status_code': 200,
+      });
+    });
+
+    test('PUT: happy path', () async {
+      final mockHub = MockHub();
+
+      final mockClient = MockClient((request) async {
+        expect(request.url, Uri.parse('https://example.com'));
+        return Response('', 200);
+      });
+
+      final client = SentryHttpClient(client: mockClient, hub: mockHub);
+
+      final response = await client.put('https://example.com');
+      expect(response.statusCode, 200);
+
+      final breadcrumb = verify(mockHub.addBreadcrumb(captureAny))
+          .captured
+          .single as Breadcrumb;
+
+      expect(breadcrumb.type, 'https');
+      expect(breadcrumb.data, <String, dynamic>{
+        'url': 'https://example.com',
+        'method': 'PUT',
+        'status_code': 200,
+      });
+    });
+
+    test('DELETE: happy path', () async {
+      final mockHub = MockHub();
+
+      final mockClient = MockClient((request) async {
+        expect(request.url, Uri.parse('https://example.com'));
+        return Response('', 200, reasonPhrase: 'OK');
+      });
+
+      final client = SentryHttpClient(client: mockClient, hub: mockHub);
+
+      final response = await client.delete('https://example.com');
+      expect(response.statusCode, 200);
+
+      final breadcrumb = verify(mockHub.addBreadcrumb(captureAny))
+          .captured
+          .single as Breadcrumb;
+
+      expect(breadcrumb.type, 'https');
+      expect(breadcrumb.data, <String, dynamic>{
+        'url': 'https://example.com',
+        'method': 'DELETE',
+        'status_code': 200,
+        'reason': 'OK',
+      });
+    });
+
+    /// Tests, that in case an exception gets thrown, that
+    ///   - no breadcrumb gets added
+    ///   - no exception gets reported by Sentry, in case the user wants to
+    ///     handle the exception
+    test('no breadcrumb for ClientException', () async {
+      final url = Uri.parse('https://example.com');
+
+      final mockHub = MockHub();
+
+      final mockClient = MockClient((request) async {
+        expect(request.url, url);
+        throw ClientException('test', url);
+      });
+
+      final client = SentryHttpClient(client: mockClient, hub: mockHub);
+
+      try {
+        await client.get('https://example.com');
+        fail('Method did not throw');
+      } on ClientException catch (e) {
+        expect(e.message, 'test');
+        expect(e.uri, url);
+      }
+
+      verifyNever(mockHub.addBreadcrumb(captureAny));
+      verifyNever(mockHub.captureException(captureAny));
+    });
+
+    /// SocketException are only a thing on dart:io platforms.
+    /// otherwise this is equal to the test above
+    test('no breadcrumb for SocketException', () async {
```

Comment:
is there a way to skip this test to the browser?

using `onPlatform: {'browser': Skip()}` maybe?

Reply:
Good point, but would't this fail anyway in browser tests because of the `dart:io` import? I'll have to look it up.

Reply:
Hm, the tests all pass on the github web pipelines. So I guess there is no need to skip this test.

Reply:
true, but looks like the CI was hanging and got canceled, mind merging the `main` branch into this one to see if CI gets happy?

Reply:
✅

Reply:
@ueman looks like the CI hangs after running the tests with `--coverage=coverage`, let me research a bit.

Reply:
not really sure about the reason, maybe related to pausing isolates for JS code, but this should be a reasonable workaround
https://github.com/getsentry/sentry-dart/pull/227

Reply:
I opened an issue on the coverage repo, so for now, just merge again the main repo into yours, sorry about that, thanks :)

Reply:
No problem